### PR TITLE
Bumped Rails, Rspec-rails, Grape and Doorkeeper

### PR DESCRIPTION
**Pull request by texpert**
Rails 5.0
Rspec-rails 3.5
Grape 0.16.2
Doorkeeper 4.0

Also implemented Travis CI matrix for exclusion of Rails 5.0 testing with Ruby < 2.2.0

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ Table of Contents
 
 ## Requirements
 - Ruby > 2.0
-- Doorkeeper > 1.4.0 and < 4.0
+- Doorkeeper > 1.4.0 and <= 4.0
```
Comment (antek-drzewiecki) [inline README.md]: This may be and < 5.0

File: wine_bouncer.gemspec [hunk 2]
```diff
@@ -18,16 +18,16 @@ Gem::Specification.new do |spec|
   spec.require_paths = ['lib']
 
   spec.add_runtime_dependency 'grape', '~> 0.10', '< 1.0'
-  spec.add_runtime_dependency 'doorkeeper', '>= 1.4', '< 4.0'
+  spec.add_runtime_dependency 'doorkeeper', '>= 1.4', '<= 4.0'
```
Comment (antek-drzewiecki) [inline wine_bouncer.gemspec]: Doorkeeper may be '< 5.0'

File: .travis.yml [hunk 3]
```diff
@@ -8,6 +8,10 @@ rvm:
 - 2.0.0
 - 2.1.0
 - 2.2.0
+- 2.2.2
+- 2.2.3
```
Comment (antek-drzewiecki) [inline .travis.yml]: Testing against 2.2.3 is not needed, otherwise the test matrix will become to large.

File: .travis.yml [hunk 4]
```diff
@@ -8,6 +8,10 @@ rvm:
...
+- 2.2.4
```
Comment (antek-drzewiecki) [inline .travis.yml]: Testing against 2.2.4 is not needed, otherwise the matrix will become to large

File: .travis.yml [hunk 5]
```diff
@@ -9,13 +9,15 @@ rvm:
 - 2.1.0
 - 2.2.0
 - 2.3.0
```
Comment (antek-drzewiecki) [inline .travis.yml]: 2.3.0 may be removed. ;)

File: Gemfile [hunk 6]
```diff
@@ -4,6 +4,8 @@ ENV['grape'] ||= '0.16.2'
 ENV['rails'] ||= '5.0.0'
 ENV['doorkeeper'] ||= '4.0.0'
 
+ruby '>=2.2.2' if ENV['rails'][0].to_i > 4
```
Comment (antek-drzewiecki) [inline Gemfile]: Not sure if this should be in the Gemfile. Rails will not run in ruby < 2.2.2 anyways.

**Comment by antek-drzewiecki** [comment]
Hey Aurel, 

Thank you for the pull request. This is a nice effort you put in!
I've added some review comments. If you can take a look at them.

Before I can merge this request I need to make sure the Travis tests were running. 
No clue why Travis hasn't tested this build. But ill look in to it.

File: Gemfile (hunk 6 above)
Comment (texpert) [inline Gemfile]: I think it is convenient when running bundle install with lower Ruby version to see the problem early.

**Comment by texpert** [comment]
Hm, maybe Travis remembered the tests from my local PR... Anyway, I will make a new commit addressing your comments and we'll see how it will go with Travis or no.

File: README.md (hunk 1 above)
Comment (texpert) [inline README.md]: Let's be not so optimistic - < 4.1 will almost certainly work without problems, while further upgrade will need to be tested prior of announcing its support :)

**Comment by texpert** [comment]
Here it goes, @antek-drzewiecki ! :)

**Comment by antek-drzewiecki** [comment]
Thank you, merged!

**Pull request merged**